### ISSUE TITLE
KoELECTRA 모델 적용 및 성능 문제 분석

### ISSUE THREAD

**Issue by eunseongy** [eunseongy/NLP-class-project_Platform-Prediction-for-News-Articles#5]
###  관련 커밋
- e9f627a0a527f781b29d6fea480d5ae07c0b48fd

요약된 뉴스 본문(summary)와 soft label을 결합하여 다음과 같은 방식으로 학습용 데이터셋을 구성함:

## 1. KoELECTRA 학습 데이터 구성
```
article_text = (
    df
    .drop_duplicates("article_id")
    .set_index("article_id")[[text_col]]
)

dataset_df = article_text.join(label_mat, how="inner").dropna()

referrer_labels = list(label_mat.columns)
num_labels = len(referrer_labels)

```

중복 기사 제거 (drop_duplicates)

article_id 기준 텍스트 + soft label 결합

Null 제거 후 학습 가능 형태로 변환

결과적으로 텍스트 요약물과 soft label이 결합된 형태로 KoELECTRA 학습에 투입됨.

## 2. KoELECTRA 모델 학습 설정

요약문은 충분히 압축되어 있으므로 KoELECTRA 입력 max_length=512 조건에 맞게 학습함.

하이퍼파라미터
```
model, tokenizer, referrers, trainer = train_model(
    df_path=summarized_excel,
    batch_size=4,
    lr=1e-5,
    num_epochs=10,
    max_length=512,
    min_ref_count=100,
)

```
lr은 1e-5일 때 가장 우수했으며 epoch가 10 이상 가면 과적합이 되는 현상이 나타나 10으로 고정하였다.

### 학습 결과 다음과 같은 문제가 발생함:

val_loss가 매우 빠르게 증가 -> 모델이 제대로 수렴하지 못함

전체적인 예측 성능도 기대 이하 (soft label 기반 예측에서 정확한 분포를 학습하지 못하는 패턴 확인)

- 미비하긴 하지만 ko-bigbird 대비 약간의 개선됨

### 내가 생각하는 원인

- 텍스트 요약만 가지고 referrer를 예측하기에는 정보가 부족할 가능성

- referrer는 기사 내용뿐 아니라 기사 배치, 채널 전략, 타겟층 등 다양한 요소와 연관됨

-> 뉴스 본문 요약 텍스트 하나만으로 이를 재현하기에는 근본적 한계가 있음

## 4. 개선 방향 제안 — Views 기반 Feature 추가 입력

현재 soft label 생성 과정에서 views(조회수) 가 중요한 역할을 했고,
EDA에서도 views와 demographics(연령/성별)과의 상관관계가 의미 있게 나타났음.

따라서 텍스트 + views 관련 feature를 함께 입력하는 멀티모달 구조 고려

- 기사 카테고리(category), 연령층 및 성별(demo) 데이터를 활용해볼 예정(기타 기사 작성 시점에 설정 가능한 매타데이터)

이러한 정보들은 기사를 작성할 때 예측할 수 있는 정보이기 때문에
플랫폼 예측이라는 목표에도 부합할 것이다.

### 기대 효과

- 텍스트만으로 부족했던 신호를 보완

- referrer의 특성과 관련된 행동/패턴 feature 반영

- val_loss 증가 문제 완화

- 모델 수렴 안정성 증가

- multi-feature 기반 예측 성능 상승 가능성

## 결론
KoELECTRA 요약 기반 학습은 빠르고 효율적이지만, 텍스트 단독으로 referrer를 예측하기에는 정보량 부족 문제가 있음.
→ 따라서 views 및 기타 메타 feature를 통합한 모델 설계로 방향 전환이 필요함.
